index.d.ts

```diff
@@ -833,7 +833,8 @@ export type ExecaChildPromise<OptionsType extends Options = Options> = {
 
 	[More info.](https://nodejs.org/api/child_process.html#subprocesskillsignal)
 	*/
-	kill(signalOrError: Parameters<ChildProcess['kill']>[0] | Error): ReturnType<ChildProcess['kill']>;
+	kill(signal: Parameters<ChildProcess['kill']>[0], error: Error): ReturnType<ChildProcess['kill']>;
+	kill(error: Error): ReturnType<ChildProcess['kill']>;
```

Comment:
```suggestion
	kill(signal: Parameters<ChildProcess['kill']>[0], error?: Error): ReturnType<ChildProcess['kill']>;
	kill(error?: Error): ReturnType<ChildProcess['kill']>;
```

?

Reply:
Right! The type tests happened to pass because this type is intersected with `& ChildProcess`, which has the definition of `.kill()` with 0 arguments, but that was by accident.